contracts/libraries/SwapIntention.sol

```diff
@@ -0,0 +1,75 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+
+enum SwapType {
+    ExactInput,
+    ExactInputSingle,
+    ExactOutput,
+    ExactOutputSingle
+}
+
+struct SwapInfo {
+    SwapType swapType;
+    bytes params;
+}
+
+struct PathKey {
+    Currency intermediateCurrency;
+    uint24 fee;
+    int24 tickSpacing;
+    IHooks hooks;
+    bytes hookData;
+}
+
+struct ExactInputSingleParams {
+    PoolKey poolKey;
+    bool zeroForOne;
+    address recipient;
+    uint128 amountIn;
+    uint160 sqrtPriceLimitX96;
+    bytes hookData;
+}
+
+struct ExactInputParams {
+    Currency currencyIn;
+    PathKey[] path;
+    address recipient;
+    uint128 amountIn;
+}
+
+struct ExactOutputSingleParams {
+    PoolKey poolKey;
+    bool zeroForOne;
+    address recipient;
+    uint128 amountOut;
+    uint160 sqrtPriceLimitX96;
+    bytes hookData;
+}
+
+struct ExactOutputParams {
+    Currency currencyOut;
+    PathKey[] path;
+    address recipient;
+    uint128 amountOut;
+    uint160 sqrtPriceLimitX96;
+}
+
+library SwapIntention {
```

Comment:
General pattern is to have struct objects and libraries supporting them to be named after the struct, so here I would have a new library file called `PathKey` and define the struct in there and any helper functions associated with the struct in a library there as well

Reply:
similar to https://github.com/Uniswap/v3-periphery/blob/main/contracts/libraries/Path.sol

Reply:
this library should go alongside PathKey in PathKe.sol and be called something more like PathKeyLib